client/landing/stepper/declarative-flow/import-flow.ts

```diff
@@ -293,6 +294,17 @@ const importFlow: Flow = {
 						return navigate( `importList?siteSlug=${ siteSlugParam }` );
 					} else if ( isMigrateFromWp && fromParam ) {
 						return navigate( `sitePicker?from=${ fromParam }` );
+					} else if (
+						urlQueryParams.has( 'showModal' ) ||
+						! isEnabled( 'migration_assistance_modal' )
+					) {
+						urlQueryParams.delete( 'showModal' );
+						return navigate( `import?siteSlug=${ siteSlugParam }` );
+					}
+
+					if ( isEnabled( 'migration_assistance_modal' ) ) {
+						urlQueryParams.set( 'showModal', 'true' );
+						return navigate( `importerWordpress?${ urlQueryParams.toString() }` );
```

Comment:
We have to be sure this happens only when migrating from a WordPress site. Should we also check with `isMigrateFromWp` in this case?

Reply:
I think this check is already happening https://github.com/Automattic/wp-calypso/pull/89812/files#diff-fce80dd6e6117b25bd509bc804a326c4c59f543c203e0312aa834ab77ab26a94R295. This logic was there before and probably for a specific case of this flow.

If `fromParam`, `isMigrateFromWp` and `show modal` are all true, the code will execute the previous logic and not the modal, which is `return navigate( 'sitePicker?from=${ fromParam }');`  I think it makes sense, right?